rules/rule.go

```diff
@@ -330,52 +346,35 @@ func (rs *ruleSet) rollupResults(id string) []RollupResult {
 }
 
 type matchedValue struct {
-	value   string
+	name    []byte
+	value   []byte
 	matched bool
 }
 
-func (rs *ruleSet) toRollupResults(id string, targets []RollupTarget) []RollupResult {
-	// Put all the tags in one map so we only need to scan the id once to get all tag values
-	numTags := 0
-	for _, t := range targets {
-		numTags += len(t.Tags)
-	}
-	allTags := make(map[string]matchedValue, numTags)
-	for _, target := range targets {
-		for _, tag := range target.Tags {
-			_, exists := allTags[tag]
-			if !exists {
-				allTags[tag] = matchedValue{}
-			}
-		}
-	}
-
-	// Find all the tag values associated with the tags in the rollup targets
-	iter := rs.iterFn(id)
-	defer iter.Close()
-	for iter.Next() {
-		name, value := iter.Current()
-		matched, exists := allTags[name]
-		if !exists {
-			continue
-		}
-		matched.value = value
-		matched.matched = true
-		allTags[name] = matched
-	}
+// toRollupResults encodes rollup target name and values into ids for each rollup target
+func (rs *ruleSet) toRollupResults(id []byte, targets []RollupTarget) []RollupResult {
+	// NB(r): This is n^2 however this should be quite fast still as
```

Comment:
Mainly because using bytes makes this need to use a map of hashed IDs which seems more expensive overall in terms of creation and computation than simply just n^2 on a small set of matched rules.